### DISCUSSION TITLE
Display nested columns as key value instead of array in SQL Lab

### DISCUSSION THREAD

**Discussion by brouberol** [apache/superset#26915] [Ideas]
Some of our Superset user have voiced that the _Cell content_ modal displaying the content of a nested column in SQL Lab isn't as helpful as could be. It is currently displayed as an array, causing them to having to go back and forth between the actual data and the schema itself.

<img width="1183" alt="Screenshot 2024-01-31 at 10 18 36" src="https://github.com/apache/superset/assets/480131/a41f3bd8-b0f3-4ed6-b076-6d75d04ffe4d">

_nested column displayed as an array_

I was wondering whether it would be valuable to have the option to display these nested columns as key/value mappings instead, such as this:
<img width="1191" alt="Screenshot 2024-01-31 at 10 18 46" src="https://github.com/apache/superset/assets/480131/32369f39-b8cd-4a2f-a975-cd374cae6ff8">

**Note**: this is the result of a small (`+15/-0`) patch applied on a staging Superset instance, in order to showcase the demo. If there's indeed interest, I'll be more than happy to open a PR to discuss the patch itself.

Thanks!

**Comment by yashAmbekar-groww** [comment]
Yes this will help alot. Can you fork and push your patch to a branch and share the branch here? Was thinking to patch superset for my org's usecase. If they add this to superset then it would be the ideal case for us.

**Comment by brouberol** [comment]
The patch can be seen [here](https://github.com/brouberol/superset/commit/de39c7626616ff6f7c326b00cfbd63688da7c704).

**Comment by mkopec87** [comment]
@brouberol , I'm also interested in that feature! Please continue with making a PR, let's see what the maintainers think :)

BTW. will your solution work for deeper nesting, that is a nested column under nested column?

**Reply by brouberol** [reply]
I'll try to take the time to submit this upstream in the near future. As it stands, I"m not sure it supports recursively nested columns. I'm sensing this might require a bit of frontend work as well. I'll investigate when I submit the patch upstream.
